spokestack/asr/cloud_client.py

```diff
@@ -137,13 +135,14 @@ def initialize(self) -> None:
 
     def disconnect(self) -> None:
         """ disconnects client socket connection """
+        self._socket.close()
```

Comment:
what happens if the client is already disconnected?
this would be a good test to write

Reply:
attribute error because it is `None`. I shouldn't let it run `disconnect` again.

Reply:
well. I mean check if the socket `is_connected` before calling close again